cli/remote_build.go

```diff
@@ -780,12 +782,27 @@ func (cmd *RemoteCmd) savePatch() error {
 func (cmd *RemoteCmd) submitBuild(project *Project, tagMap map[string]string) error {
 
 	startTime := time.Now()
-	var submitProgress *Progress
+	var submitProgress, remoteProgress *Progress
 	submitErrored := func() {
 		if submitProgress != nil {
 			submitProgress.Fail()
 		}
 	}
+	remoteErrored := func() {
+		if remoteProgress != nil {
+			remoteProgress.Fail()
+		}
+	}
+
+	c := make(chan os.Signal)
+	signal.Notify(c, os.Interrupt, syscall.SIGTERM)
+	go func() {
```

Comment:
This goroutine worries me for a few reasons:

1. The goroutine will leak past the end of `submitBuild` if an interrupt signal is not sent to the process.
2. The calls to `submitErrored` and `remoteErrored` both race on the assignment to these local variables.
3. It races with indicating success: imagine the case where the build succeeds and then an interrupt signal is sent before the main goroutine is able to report success. It's undefined which one of these would be called first.

I recommend that we start plumbing `Context` throughout the `yb` codebase, which is the standard pattern in Go for dealing with these sorts of issues. See https://talks.golang.org/2014/gotham-context.slide for a brief overview.